code/modules/bitrunning/components/avatar_gear.dm

```diff
@@ -0,0 +1,67 @@
+
+/// Tracks the highest up human carrying this item, and loads bitrunning gear onto their avatar when they enter a virtual domain
+/datum/component/loads_avatar_gear
+	/// The callback called when COMSIG_BITRUNNER_STOCKING_GEAR is sent to our host human
+	var/datum/callback/load_callback
+	/// The human we are currently carried by
+	var/mob/living/carbon/human/tracked_human
+
+/datum/component/loads_avatar_gear/Initialize(datum/callback/load_callback)
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.load_callback = load_callback
+
+/datum/component/loads_avatar_gear/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ATOM_ENTERING, PROC_REF(on_entered_loc))
+
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERING = PROC_REF(on_entered_loc),
+	)
+	AddComponent(/datum/component/connect_containers, parent, loc_connections)
+
+/datum/component/loads_avatar_gear/UnregisterFromParent()
+	UnregisterSignal(parent, list(
+		COMSIG_ATOM_ENTERING,
+	))
+
+	qdel(GetComponent(/datum/component/connect_containers))
+
+/datum/component/loads_avatar_gear/Destroy(force)
+	load_callback = null
+	return ..()
+
+
+/datum/component/loads_avatar_gear/proc/on_entered_loc(datum/source, atom/destination, atom/old_loc, list/atom/old_locs)
+	SIGNAL_HANDLER
+
+	// No need to do checks if we're just moving turfs
+	if(isturf(destination) && isturf(old_loc))
+		return
+
+	// Iterate over list in reverse so we get the topmost human first
+	var/list/nested_locs = get_nested_locs(parent)
+	for(var/i = length(nested_locs), i > 0, i--)
+		var/atom/container = nested_locs[i]
+		if(ishuman(container))
+			start_tracking(container)
+			return
```

Comment:
why the topmost human?
if above is unsure, would we prefer get() here?

Reply:
also optionally use `for(var/i in max to min step -1)`, dealer's choice

Reply:
So I thought about this initially, and the biggest difference between both orders is that when we consider an `avatar_gear` item carried by someone inside a bluespace bodybag carried by our bitrunner:
1. We pick bottommost, it doesn't do anything
2. We pick topmost, they get to use it
And I think the second is much funnier than the former